### PR TITLE
KUZBO-129 Authenticated session

### PR DESCRIPTION
**Pull request by jenow**
- Implemented Login with cookie
- Wrote unit tests

TODO:
Write e2e tests
Implement material design when the global-layout branch is finished

### CODE REVIEW

File: src/services/cookies.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export default {
```
Comment (xbill82) [inline src/services/cookies.js]: I have nothing against cookies, but @AnthonySendra leveraged the SessionStorage in the Klack App, wouldn't it be better? Or even LocalStorage if we want real persistence...
As far as I understand from [this thread](http://stackoverflow.com/questions/3220660/local-storage-vs-cookies), Cookies are essentially useful when their content must be sent to the server at every request (which is not our case).
That being said, this module is useful and should not be dropped even if we choose another method to implement the session.

File: test/e2e/specs/test.js [hunk 2]
```diff
@@ -4,11 +4,9 @@
 module.exports = {
   'default e2e tests': function (browser) {
     browser
-    .url('http://localhost:8080')
-      .waitForElementVisible('#app', 5000)
-      .assert.elementPresent('.logo')
-      .assert.containsText('h1', 'Hello World!')
-      .assert.elementCount('p', 3)
+    .url('http://localhost:3000')
```
Comment (xbill82) [inline test/e2e/specs/test.js]: Would be nice to get this address from a config file.

File: test/unit/specs/auth/component.spec.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import Vue from 'vue'
+import Login from '../../../../src/components/Login'
+
+describe('Login.vue', () => {
+  it('should render correct contents', () => {
```
Comment (xbill82) [inline test/unit/specs/auth/component.spec.js]: This title is copy pasted from somwhere, I think

File: test/unit/specs/auth/component.spec.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+    expect(Login.data().username).to.be.null
+    expect(Login.data().password).to.be.null
+    expect(Login.vuex.actions.doLogin).to.be.defined
+  })
+})
```
Comment (xbill82) [inline test/unit/specs/auth/component.spec.js]: So we don't test the rendered UI here? Do we leave it to integration tests?

File: test/unit/specs/auth/mutations.spec.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { expect } from 'chai'
+import { mutations } from '../../../../src/vuex/modules/auth/store'
+
+const { SET_CURRENT_USER, SET_CURRENT_USER_RIGHTS } = mutations
+
+describe('auth mutations test', () => {
+  it('should set the user status', () => {
+    let stat = {user: null}
```
Comment (xbill82) [inline test/unit/specs/auth/mutations.spec.js]: Is `stat` instead of `state` on purpose?

**Comment by xbill82** [comment]
Great PR. The tests are just GREAT!

File: src/components/Login.vue [hunk 6]
```diff
@@ -23,7 +23,9 @@ export default {
     }
   },
   vuex: {
-    actions: { doLogin },
+    actions: {
+      doLogin
```
Comment (jenow) [inline src/components/Login.vue]: I agree, I will call `logout` `doLogout`

File: test/unit/specs/auth/component.spec.js (hunk 4 above)
Comment (jenow) [inline test/unit/specs/auth/component.spec.js]: We should, I wrote it in the TODO

**Comment by AnthonySendra** [comment]
Ok for me, but the merge must be done on branch develop instead of master ;)

**Comment by jenow** [comment]
New PR to develop branch

**Pull request closed without merging**